doc/reference.rst

```diff
@@ -447,6 +485,56 @@ The following atom types are available:
 - ``Chemfiles.DUMMY_ATOM``: Dummy site, with no physical reality
 - ``Chemfiles.UNDEFINED_ATOM``: Undefined atom type
 
+.. _Residue:
+
+``Residue`` type and associated function
+------------------------------------------
+
+.. jl:function:: Residue(name::String, resid::Integer)
+
+    Create a new residue with the given `name` and residue identifier `resid`.
+
+.. jl:function:: Residue(name::String)
+
+    Create a new residue with the given `name`.
+
+.. jl:function:: Residue(topology::Topology, index::Integer)
+
+    Get a copy of the residue at `index` from a `topology`.
+    If `index` is bigger than the result of `count_residues`, this function will return `nothing`.
+    The residue index in the topology is not always the same as the residue `id`.
+
+.. jl:function:: residue_for_atom(topology::Topology, index::Integer)
+
+    Get a copy of the residue containing the atom at `index` in the `topology`.
+    This function will return `nothing` if the atom is not in a residue, or if the
+    `index` is bigger than `natoms`.
+
+.. jl:function:: name(residue::Residue)
+
+    Get the name of a `residue`.
+
+.. jl:function:: id(residue::Residue)
+
+    Get the identifier of a `residue` in the initial topology.
+
+.. jl:function:: id(residue::Residue)
+
+    Get the identifier of a `residue` in the initial topology.
+
+.. jl:function:: natoms(residue::Residue)
```

Comment:
We could also just remove the natoms function. What do you think ?

I putted it there to mimic the C API, but it does not seems worth it to have the same function twice ...